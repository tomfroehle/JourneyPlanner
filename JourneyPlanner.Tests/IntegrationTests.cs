using Snapshooter.NUnit;

namespace JourneyPlanner.Tests;

public class IntegrationTests
{
    [Test]
    public void Should_Match_Snapshot()
    {
        var journeys = JourneySearcher.FindShortestJourneys("Jungfernsteig", "Hauptbahnhof");

        journeys.MatchSnapshot();
    }

    [Test]
    public void Should_Be_Empty_If_There_Are_No_Routes()
    {
        var journeys = JourneySearcher.FindShortestJourneys("non-existing", "Hauptbahnhof");

        CollectionAssert.IsEmpty(journeys);
    }
}