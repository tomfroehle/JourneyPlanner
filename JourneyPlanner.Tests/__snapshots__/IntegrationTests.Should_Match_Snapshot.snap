[
  {
    "Connections": [
      {
        "Source": {
          "Name": "Jungfernsteig",
          "Line": "U2"
        },
        "Destination": {
          "Name": "Hauptbahnhof",
          "Line": "U2"
        },
        "Duration": 2
      }
    ],
    "Duration": 2
  },
  {
    "Connections": [
      {
        "Source": {
          "Name": "Jungfernsteig",
          "Line": "U1 Nord"
        },
        "Destination": {
          "Name": "Stephansplatz",
          "Line": "U1 Nord"
        },
        "Duration": 2
      },
      {
        "Source": {
          "Name": "Stephansplatz",
          "Line": "U1 Nord"
        },
        "Destination": {
          "Name": "Hauptbahnhof",
          "Line": "U1 Nord"
        },
        "Duration": 3
      }
    ],
    "Duration": 5
  },
  {
    "Connections": [
      {
        "Source": {
          "Name": "Jungfernsteig",
          "Line": "U3"
        },
        "Destination": {
          "Name": "Hauptstraße",
          "Line": "U3"
        },
        "Duration": 2
      },
      {
        "Source": {
          "Name": "Hauptstraße",
          "Line": "U3"
        },
        "Destination": {
          "Name": "Stephansplatz",
          "Line": "U3"
        },
        "Duration": 2
      },
      {
        "Source": {
          "Name": "Stephansplatz",
          "Line": "U3"
        },
        "Destination": {
          "Name": "Stephansplatz",
          "Line": "U1 Nord"
        },
        "Duration": 0
      },
      {
        "Source": {
          "Name": "Stephansplatz",
          "Line": "U1 Nord"
        },
        "Destination": {
          "Name": "Hauptbahnhof",
          "Line": "U1 Nord"
        },
        "Duration": 3
      }
    ],
    "Duration": 7
  }
]
